src/index.js

```diff
@@ -21,20 +21,59 @@ function crossEnv(args, options = {}) {
         env,
       },
     )
-    process.on('SIGTERM', () => proc.kill('SIGTERM'))
-    process.on('SIGINT', () => proc.kill('SIGINT'))
-    process.on('SIGBREAK', () => proc.kill('SIGBREAK'))
-    process.on('SIGHUP', () => proc.kill('SIGHUP'))
-    proc.on('exit', (code, signal) => {
-      let crossEnvExitCode = code
-      // exit code could be null when OS kills the process(out of memory, etc) or due to node handling it
-      // but if the signal is SIGINT the user exited the process so we want exit code 0
-      if (crossEnvExitCode === null) {
-        crossEnvExitCode = signal === 'SIGINT' ? 0 : 1
+
+    // See https://github.com/jtlapp/node-cleanup
+    //
+    // > When you hit Ctrl-C, you send a SIGINT signal to each process in the
+    // > current process group. A process group is set of processes that are
+    // > all supposed to end together as a group instead of persisting
+    // > independently. However, some programs, such as Emacs, intercept and
+    // > repurpose SIGINT so that it does not end the process. In such cases,
+    // > SIGINT should not end any processes of the group.
+    //
+    // Delegate decision to terminate to child process, if the child exits on
+    // SIGINT then the `child.on('exit')` callback will be invoked, re-raising
+    // the signal to the parent process
+    const delegateSignalToChild = signal => () => {
+      // SIGINT is sent to all processes in group, no need to delegate.
+      if (child && signal !== 'SIGINT') {
+        process.kill(child.pid, signal)
+      }
+    }
+
+    const sigtermHandler = delegateSignalToChild('SIGTERM')
```

Comment:
The signal event receives the signal name as the argument. That means you don't need to create a special handler for each signal:
```js
const delegateSignalToChild = signal => {
  process.kill(child.pid, signal);
}
process.on('SIGTERM', delegateSignalToChild);
process.on('SIGBREAK', delegateSignalToChild);
```

Reply:
Sure. I was following other examples around the place to get the correct handling working.

As the original code created inline arrow functions to handle the individual events I just kept a similar code flow.

I wasn't sure whether the listeners would vary or be removed independently so went with the more verbose style.